### PR TITLE
Display SlimeVR Tracker Battery Life

### PR DESCRIPTION
**Pull request by NWBx01**
Adds support for receiving tracker battery info from the SlimeVR Server.
- Trackers that do not have battery info are not affected.
- Trackers that do have battery info will report their battery percentage, as well as whether they're charging or not.
- Works for owoTrack devices (However, these will not display whether they're charging, as owoTrack does not send battery voltage info)

This depends on another pull request. (https://github.com/SlimeVR/SlimeVR-Server/pull/668)

### CODE REVIEW

**Comment by kitlith** [comment]
The only thing battery voltage is used for here is for detecting if the tracker is charging or not, correct? I'd rather kick charging detection into firmware (or at least server for now) (i.e. the device reports the right thing depending on the setup) and just receive a boolean is_charging in driver.

File: src/VRDriver.cpp [hunk 1]
```diff
@@ -167,6 +173,16 @@ void SlimeVRDriver::VRDriver::RunFrame()
...
         sendBridgeMessage(*message, *this);
+
+        vr::ETrackedPropertyError err;
+        if (vr::VRProperties()->GetBoolProperty(vr::VRProperties()->TrackedDeviceToPropertyContainer(0), vr::Prop_DeviceProvidesBatteryStatus_Bool, &err) == true) {
+            messages::Battery* hmdBattery = google::protobuf::Arena::CreateMessage<messages::Battery>(&arena);
+            message->set_allocated_battery(hmdBattery);
+            hmdBattery->set_tracker_id(0);
+            hmdBattery->set_battery_level(vr::VRProperties()->GetFloatProperty(vr::VRProperties()->TrackedDeviceToPropertyContainer(0), vr::Prop_DeviceBatteryPercentage_Float, &err) * 100);
+            hmdBattery->set_is_charging(vr::VRProperties()->GetBoolProperty(vr::VRProperties()->TrackedDeviceToPropertyContainer(0), vr::Prop_DeviceIsCharging_Bool, &err));
+            sendBridgeMessage(*message, *this);
```
Comment (kitlith) [inline src/VRDriver.cpp:184]: I found this unexpected, but if SlimeVR Server uses/relays the HMD battery level, then that's fine.

**Review by kitlith — APPROVED** [review]

File: src/VRDriver.cpp (hunk 1 above)
Comment (NWBx01) [reply src/VRDriver.cpp:184]: It's mostly just a little cosmetic thing. I thought about doing the same for other devices like controllers and Vive trackers, but I think that's something for the Feeder App.

SlimeVR 5_22_2023 6_56_51 PM

Comment (kitlith) [reply src/VRDriver.cpp:184]: If you could make an issue on the feeder app repo to remind me to send these messages, that would be appreciated <3

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**